lib/archethic/contracts/interpreter/library/common/http_impl.ex

```diff
@@ -0,0 +1,156 @@
+defmodule Archethic.Contracts.Interpreter.Library.Common.HttpImpl do
+  @moduledoc """
+  Http client for the Smart Contracts.
+  Implements AEIP-20.
+
+  Mint library is processless so in order to not mess with
+  other processes, we use it from inside a Task.
+  """
+
+  alias Archethic.Contracts.Interpreter.Library
+  alias Archethic.Contracts.Interpreter.Library.Common.Http
+  alias Archethic.TaskSupervisor
+
+  @behaviour Http
+  @threshold 256 * 1024
+
+  @impl Http
+  def fetch(uri) do
+    task =
+      Task.Supervisor.async_nolink(
+        TaskSupervisor,
+        fn -> do_fetch(uri) end
+      )
+
+    case Task.yield(task, 2_000) || Task.shutdown(task) do
+      {:ok, {:ok, reply}} ->
+        reply
+
+      {:ok, {:error, :threshold_reached}} ->
+        raise Library.Error, message: "Http.fetch/1 response is bigger than threshold"
+
+      {:ok, {:error, _}} ->
+        # Mint.HTTP.connect error
+        # Mint.HTTP.stream error
+        raise Library.Error, message: "Http.fetch/1 failed"
+
+      {:ok, {:error, _, _}} ->
+        # Mint.HTTP.request error
+        raise Library.Error, message: "Http.fetch/1 failed"
+
+      nil ->
+        # Task.shutdown
+        raise Library.Error, message: "Http.fetch/1 timed out for url: #{uri}"
+    end
+  end
+
+  @impl Http
+  def fetch_many(uris) do
+    uris_count = length(uris)
+
+    if uris_count > 5 do
+      raise Library.Error, message: "Http.fetch_many/1 was called with too many urls"
+    else
+      Task.Supervisor.async_stream_nolink(
+        TaskSupervisor,
+        uris,
+        &fetch/1,
```

Comment:
I think you should no use fetch here since it create a Task whereas `async_stream_nolink` already create tasks. So here you have nested tasks while you could afford it.
You could directly use `do_fetch` and specify timeout in `async_no_link` with a `timeout: 2000` and `on_timeout: :kill_task`